buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java

```diff
@@ -177,4 +189,29 @@ private String readFile(Path path) {
   @Test public void processFileMultipleFunctionsSeparatedByLines() {
     assertFileProcessed("multiple_functions_separated");
   }
+
+  @Test public void processFileTokenAssignments() {
+    assertFileProcessed("token_assignments");
+  }
+
+  @Test public void processFileFunctionsAndAssignments() {
+    assertFileProcessed("functions_and_assignments");
+  }
+
+  @Test public void processTokenAssignmentEmpty() {
+    String declaration = "TOKEN :\n";
+    String assignment = declaration
+      + "{\n}";
+    assertTokenAssignmentIsParsed(declaration, assignment);
+  }
+
+  @Test public void processTokenAssignmentNonEmpty() {
```

Comment:
Should we add unit test cases for MORE and SKIP token assignments?

Reply:
Will add.